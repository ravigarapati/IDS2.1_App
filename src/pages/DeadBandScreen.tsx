import React, {useEffect, useState} from 'react';
import {View, StyleSheet, Text, ScrollView, Image} from 'react-native';
import OptionText from '../components/OptionText';
import {
  BoschIcon,
  CustomText,
  Button,
  ProgressIndicator,
  SwitchContent,
  CustomWheelPicker,
} from '../components';
import {Icons} from '../utils/icons';
import {Colors} from '../styles';
import {Dictionary} from '../utils/dictionary';
import {
  getAdvancedSettings,
  getDeviceInformation,
  lockScreen,
  setAdccessSettings,
} from '../store/actions/HomeOwnerActions';
import {connect, useDispatch} from 'react-redux';
import THERMOMETER from '../assets/images/temperature1.svg';
import { CustomWheelPick } from '../components/CustomWheelPick';

function DeadBandScreen({
  navigation,
  getDeviceInformation,
  deviceInformation,
  selectedDevice,
}) {
  const dispatch = useDispatch();
  const [deadband, setDeadband] = useState('');
  const [deadbandValues, setDeadbandValues] = useState(
    selectedDevice.isFahrenheit
      ? Array.from({length: (9 - 2) / 1 + 1}, (_, i) => `${2 + i * 1} 째F`)
      : Array.from({length: (5 - 1) / 1 + 1}, (_, i) => `${1 + i * 1} 째C`),
  );

  const [values, setValues] = useState({});

  useEffect(() => {
    loadRunTimeValues();
  }, []);

  function loadRunTimeValues() {
    return new Promise<void>((resolve, reject) => {
      dispatch(
        getAdvancedSettings(selectedDevice.macId, response => {
          const db = response.db;
          const hc = response.hc;
          const hsd = response.hsd;
          const latching = response.latching;
          const sc = response.sc;
          const ss = response.ss;
          const stageDelay = response.stageDelay;
          const stageTemp = response.stageTemp;

          setValues({
            db,
            hc,
            hsd,
            latching,
            sc,
            ss,
            stageDelay,
            stageTemp,
          });

          setDeadband(db);

          resolve();
        }),
      );
    });
  }
  function setAdvancedSettings(value) {
    return new Promise<void>((resolve, reject) => {
      dispatch(
        setAdccessSettings(selectedDevice.macId, value, response => {
          resolve();
        }),
      );
    });
  }

  const setDb = async selected => {
    let replacementArray =
      deadbandValues[selected ? selected['0'] : 0].split(' ');
    let replacement = replacementArray[0];
    setDeadband((replacement * 10).toString());
    setValues({...values, db: replacement});
    await setAdvancedSettings({...values, db: (replacement * 10).toString()});
    dispatch({
      type: 'SHOW_LOADING',
      data: true,
    });
    setTimeout(() => {
      loadRunTimeValues();
    }, 2000);
  };

  return (
    <View
      style={{
        flex: 1,
        backgroundColor: 'white',
        justifyContent: 'space-between',
      }}>
      <Image
        style={{height: 8, width: '100%'}}
        source={require('../assets/images/header_ribbon.png')}
      />
      <ScrollView>
        <View>
          <View style={{padding: 15}}>
            <View>
              <CustomWheelPick
                type={'picker'}
                testID={'setDeadBand'}
                edit={true}
                blur={true}
                accessibilityHintText={
                  'Activate to open the modal to select the deadband.'
                }
                pickerWidth={'100%'}
                placeholder={'Set to'}
                value={
                  deadband / 10 + (selectedDevice.isFahrenheit ? ' 째F' : ' 째C')
                }
                isRequiredField={true}
                values={deadbandValues}
                isSvgIcon={true}
                icon={<THERMOMETER fill="#000" />}
                onConfirm={selected => {
                  setDb(selected);
                }}
                accessibilityWheelPickerValue={[
                  ' the amount of degrees for deadband.',
                ]}
                defaultIndex={0}
                defaultValue={deadbandValues[0]}
              />
            </View>
            <View style={styles.tipSection}>
              <BoschIcon
                size={20}
                name={Icons.infoTooltip}
                color={Colors.mediumBlue}
                accessibilityLabel={'Info'}
              />
              <CustomText
                allowFontScaling={true}
                accessibilityLabelText={
                  'Use the deadband to adjust the minimum value allowed between the heating & cooling set points in Auto mode.'
                }
                size={12}
                align="left"
                newline={true}
                text={
                  'Use the deadband to adjust the minimum value allowed between the heating & cooling set points in Auto mode.'
                }
                style={[styles.flexShrink1, styles.paddingLeft5]}
              />
            </View>
          </View>
        </View>
      </ScrollView>
    </View>
  );
}

const mapStateToProps = state => {
  return {
    deviceStatus: state.homeOwner.deviceInformation,
    selectedDevice: state.homeOwner.selectedDevice,
  };
};

const mapDispatchToProps = {
  getDeviceInformation,
};

export default connect(mapStateToProps, mapDispatchToProps)(DeadBandScreen);

const styles = StyleSheet.create({
  optionContainer: {
    borderBottomWidth: 1,
    borderColor: '#BFC0C2',
  },
  tipSection: {
    flexDirection: 'row',
    marginTop: 33,
  },
  flexShrink1: {
    flexShrink: 1,
  },
  paddingLeft5: {
    paddingLeft: 12,
  },
});
