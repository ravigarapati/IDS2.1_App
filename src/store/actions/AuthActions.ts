import AsyncStorage from '@react-native-async-storage/async-storage';
import {
  CURRENT_USER,
  CURRENT_STEP,
  UPDATE_CURRENT_USER,
  LOGOUT,
  TERMS_AND_CONDITIONS,
  LATEST_VERSION,
  SET_STEP_TITLES,
  SET_USER_ROLE,
  SET_SHOW_ROLE,
  SET_USER_ADDRESS,
  RESET_USER,
  SET_USER_NAME,
} from '../labels/AuthLabels';
import {Auth} from 'aws-amplify';
import {showToast} from '../../components/CustomToast';
import {
  apiInterceptorForPost,
  apiInterceptorForPostCustomApi,
  openApiInterceptorForGet,
  apiInterceptorForPostAPI,
} from './CommonActions';
import ClearCache from 'oa-react-native-clear-cache';

export const setCurrentUser = user => {
  return async dispatch => {
    dispatch({type: CURRENT_USER, user: user});
    ClearCache.clearAppCache(() => {});
  };
};
export const setCurrentStep = stepNumber => {
  return async dispatch => {
    dispatch({type: CURRENT_STEP, data: stepNumber});
  };
};
export const updateUserAttributes = (userFromStore, data, callback?) => {
  return async dispatch => {
    Auth.updateUserAttributes(userFromStore, data)
      .then(() => {
        if (callback) {
          callback();
        }
      })
      .catch(error => {
        showToast(error.message, 'error');
      });
  };
};

export const verifyCurrentUserAttribute = attribute => {
  Auth.verifyCurrentUserAttribute(attribute)
    .then(() => {})
    .catch(error => {
      showToast(error.message, 'error');
    });
};

export const verifyCurrentUserAttributeSubmit = (
  attribute,
  value,
  callback?,
) => {
  Auth.verifyCurrentUserAttributeSubmit(attribute, value)
    .then(() => {
      if (callback) {
        callback();
      }
    })
    .catch(error => {
      showToast(error.message, 'error');
    });
};

export const updateUserObject = attributes => {
  return async dispatch => {
    dispatch({type: UPDATE_CURRENT_USER, data: attributes});
  };
};

export const checkAdminDetails = (attr, callback?) => {
  return async dispatch => {
    const path = '/checkadmindetails';
    apiInterceptorForPost(
      dispatch,
      path,
      attr,
      response => {
        callback();
      },
      errorResponse => {
        showToast(errorResponse, 'error');
      },
    );
  };
};

export const createProfile = (userData, callback?) => {
  return async dispatch => {
    const path = '/registration';
    apiInterceptorForPost(
      dispatch,
      path,
      userData,
      response => {
        callback();
      },
      errorResponse => {
        showToast(errorResponse, 'error');
      },
    );
  };
};

export const createProfileHomeOwner = (userData, callback?) => {
  return async dispatch => {
    const baseUrl =
      'https://r9nzt6y3ui.execute-api.us-east-1.amazonaws.com/development';
    apiInterceptorForPostCustomApi(
      dispatch,
      baseUrl,
      '/user/register',
      {
        firstName: userData.firstName,
        lastName: userData.lastName,
        phoneNumber: userData.phoneNumber,
        type: 'HomeOwner',
        timestamp: `${new Date().valueOf()}`,
      },
      response => {
        if (response.message) {
          if (
            response.message === 'Operation succeed' ||
            response.message.startsWith("'e")
          ) {
            callback();
          } else if (
            response.message === 'The user account has already registered'
          ) {
            showToast(response.message, 'info');
          } else {
            showToast(response.message, 'error');
          }
        }
      },
      errorResponse => {
        showToast(errorResponse, 'error');
      },
    );
  };
};

export const logout = () => {
  return async dispatch => {
    dispatch({type: LOGOUT, data: {}});
    // clear the storage
    ClearCache.getAppCacheSize(data => {});
    ClearCache.clearAppCache(() => {});
    await AsyncStorage.removeItem('persist:root').then(() => {});
  };
};

export const getTermsandcondition = () => {
  return async dispatch => {
    const path = '/termsandconditions';
    openApiInterceptorForGet(
      dispatch,
      path,
      response => {
        dispatch({type: TERMS_AND_CONDITIONS, data: response.data});
      },
      errorResponse => {
        showToast(errorResponse, 'error');
      },
    );
  };
};

export const updateTermsAndConditions = (input, successCallback) => {
  return async dispatch => {
    const path = '/termsandconditions';
    apiInterceptorForPost(
      dispatch,
      path,
      input,
      response => {
        successCallback();
      },
      errorResponse => {
        showToast(errorResponse, 'error');
      },
    );
  };
};

export const deleteUser = (userRole, successCallback) => {
  return async dispatch => {
    const path = `/deleteuser/${userRole.type}`;
    apiInterceptorForPost(
      dispatch,
      path,
      {},
      response => {
        successCallback();
      },
      errorResponse => {
        showToast(errorResponse, 'error');
      },
    );
  };
};

export const getLatestVersion = (successCallback?: any) => {
  return async dispatch => {
    const path = '/latestappversion';
    openApiInterceptorForGet(
      dispatch,
      path,
      response => {
        dispatch({type: LATEST_VERSION, data: response.data});
        successCallback && successCallback(response.data);
      },
      errorResponse => {
        showToast(errorResponse, 'error');
      },
    );
  };
};

export const setStepTitles = stepTitles => {
  return async dispatch => {
    dispatch({type: SET_STEP_TITLES, data: stepTitles});
  };
};

export const setUserRole = userRole => {
  return async dispatch => {
    dispatch({type: SET_USER_ROLE, data: userRole});
  };
};

export const setShowRole = showRole => {
  return async dispatch => {
    dispatch({type: SET_SHOW_ROLE, data: showRole});
  };
};

export const setUserName = userName => {
  return async dispatch => {
    dispatch({type: SET_USER_NAME, data: userName});
  };
};

export const setUserAddress = userAddress => {
  return async dispatch => {
    dispatch({type: SET_USER_ADDRESS, data: userAddress});
  };
};

export const resetUser = () => {
  return async dispatch => {
    dispatch({type: RESET_USER});
  };
};