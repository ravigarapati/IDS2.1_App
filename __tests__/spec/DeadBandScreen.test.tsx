import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import {StyleSheet} from 'react-native';
import DeadBandScreen from './../../src/pages/DeadBandScreen';
import {render, cleanup, fireEvent} from 'react-native-testing-library';
import * as homeOwnerActions from './../../src/store/actions/HomeOwnerActions';
import thunk from 'redux-thunk';
const middlewares = [thunk];
import {createStore, combineReducers, applyMiddleware} from 'redux';
import HomeOwnerReducer from './../../src/store/reducers/HomeOwnerReducer';
import * as apiCalls from './../../src/store/actions/CommonActions';

const rootReducer = combineReducers({
  homeOwner: HomeOwnerReducer,
});

const realStore = createStore(
  rootReducer,
  {
    homeOwner: {
      energyData: {},
      deviceList: [],
      selectedDevice: {
        lockDevice: false,
        d_hour: '1',
        isFahrenheit: true,
        deadband: 3,
        schedules: [
          {
            default_home: '1',
            limit: '71-81',
            mode: '1',
            model_id: '1',
            name: 'Home',
            state: '0',
          },
          {
            default_home: '1',
            limit: '71-81',
            mode: '1',
            model_id: '2',
            name: 'Vacation',
            state: '0',
          },
          {
            default_home: 0,
            limit: '71-82',
            mode: '3',
            model_id: '3',
            name: 'summer ',
            state: '1',
          },
          {
            default_home: 0,
            limit: '71-82',
            mode: '3',
            model_id: '4',
            name: 'summer cam',
            state: '0',
          },
        ],
      },
      contractorMonitoringStatus: false,
      fanMode: '',
      fanOnFor: '',
      fanOffFor: '',
      fanIsScheduled: '',
      fanScheduledStart: '',
      fanScheduledEnd: '',
      selectedSchedule: '1',
      selectedScheduleName: '',
      scheduleInfo: {
        items1: [{c: '0', h: '0', t: '78.0-70.0'}],
        items2: [
          {c: '0', h: '0', t: '78.0-73.0'},
          {c: '0', h: '0', t: '99.0-90.0'},
        ],
        items3: [{c: '0', h: '1185', t: '52.0-48.0'}],
        items4: [{c: '0', h: '0', t: '78.0-70.0'}],
        items5: [{c: '0', h: '0', t: '78.0-70.0'}],
        items6: [{c: '0', h: '0', t: '78.0-70.0'}],
        items7: [{c: '0', h: '0', t: '78.0-70.0'}],
      },
      prevSelectedDevice: {},
      deviceStatus: {},
      deviceInformation: {},
      deviceDetails: {},
      heatPumpInfo: {},
      reloadHeatPumpInfo: false,
      notificationList: [],
      lastKey: {},
      notificationsCount: 0,
      lastUpdated: null,
      isTermsConditionsAccepted: null,
      hoUserAnalytics: null,
      selectedUnitName: {},
      isThermostatSelected: true,
      isUserFirstLogin: true,
      actualWeatherOnFahrenheit: true,
      hapticVibration: true,
      _24HrsFormatSelected: false,
      location: 'Alabaster',
      updatedSelectedDevice: false,
      weatherInfo: {},
      weatherInfoLocation: {},
      locationSuggestions: [
        {
          PlaceId:
            'AQABAFMAKZmJqReTuJvQ0SYSvNXBFPnTNre8NpZN_vO5ppVBiIVKMWyFo7U_23WZIEB3jBaCX9aLHNHy6gHy6QyfVv1o8QfoXctScRaEWzssnUy2A7vSsswLy5wkja6Jfp6hQXlv4QvGMRLRNcLPAogsltpxpywjew',
          Text: 'New York, NY, United States',
        },
      ],
      locationInformation: {},
      idsSelectedDeviceAccess: '',
      utilyEnergySaving: '',
      idsSelectedDevice: '',
      idsSelectedOdu: '',
      idsSelectedDeviceType: '',
      tempLocation: '',
      newDeviceInfo: {},
      previousBcc: '',
      deviceList2: [
        /*{
          id: 2,
          //macId: 'e8fdf8a4773c',
          macId: '34eae7c351e6',
          deviceName: 'Bedroom Thermostat',
          deviceType: 'BCC100',
          isThermostat: true,
          isOn: true,
          mode: 0,
          setPoint: 74,
          heatingSetpoint: 74,
          coolingSetpoint: 79,
          current: 76,
          isAccesories: true,
          accesories: [
            {
              id: 0,
              name: 'Dehumidifier',
              isEnable: 0,
              isOn: false,
            },
            {
              id: 1,
              name: 'Humidifier',
              isEnable: 1,
              isOn: false,
            },
          ],
          isMonitoring: null,
          isOnSchedule: true,
          stage: 3,
          isConnected: true,
          acceleratedHeating: true,
          schedules: [],
        },*/
      ],
    },
  },
  applyMiddleware(thunk),
);

const realStoreCelcius = createStore(
  rootReducer,
  {
    homeOwner: {
      energyData: {},
      deviceList: [],
      selectedDevice: {
        lockDevice: false,
        d_hour: '1',
        isFahrenheit: false,
        deadband: 3,
        schedules: [
          {
            default_home: '1',
            limit: '71-81',
            mode: '1',
            model_id: '1',
            name: 'Home',
            state: '0',
          },
          {
            default_home: '1',
            limit: '71-81',
            mode: '1',
            model_id: '2',
            name: 'Vacation',
            state: '0',
          },
          {
            default_home: 0,
            limit: '71-82',
            mode: '3',
            model_id: '3',
            name: 'summer ',
            state: '1',
          },
          {
            default_home: 0,
            limit: '71-82',
            mode: '3',
            model_id: '4',
            name: 'summer cam',
            state: '0',
          },
        ],
      },
      contractorMonitoringStatus: false,
      fanMode: '',
      fanOnFor: '',
      fanOffFor: '',
      fanIsScheduled: '',
      fanScheduledStart: '',
      fanScheduledEnd: '',
      selectedSchedule: '1',
      selectedScheduleName: '',
      scheduleInfo: {
        items1: [{c: '0', h: '0', t: '78.0-70.0'}],
        items2: [
          {c: '0', h: '0', t: '78.0-73.0'},
          {c: '0', h: '0', t: '99.0-90.0'},
        ],
        items3: [{c: '0', h: '1185', t: '52.0-48.0'}],
        items4: [{c: '0', h: '0', t: '78.0-70.0'}],
        items5: [{c: '0', h: '0', t: '78.0-70.0'}],
        items6: [{c: '0', h: '0', t: '78.0-70.0'}],
        items7: [{c: '0', h: '0', t: '78.0-70.0'}],
      },
      prevSelectedDevice: {},
      deviceStatus: {},
      deviceInformation: {},
      deviceDetails: {},
      heatPumpInfo: {},
      reloadHeatPumpInfo: false,
      notificationList: [],
      lastKey: {},
      notificationsCount: 0,
      lastUpdated: null,
      isTermsConditionsAccepted: null,
      hoUserAnalytics: null,
      selectedUnitName: {},
      isThermostatSelected: true,
      isUserFirstLogin: true,
      actualWeatherOnFahrenheit: true,
      hapticVibration: true,
      _24HrsFormatSelected: false,
      location: 'Alabaster',
      updatedSelectedDevice: false,
      weatherInfo: {},
      weatherInfoLocation: {},
      locationSuggestions: [
        {
          PlaceId:
            'AQABAFMAKZmJqReTuJvQ0SYSvNXBFPnTNre8NpZN_vO5ppVBiIVKMWyFo7U_23WZIEB3jBaCX9aLHNHy6gHy6QyfVv1o8QfoXctScRaEWzssnUy2A7vSsswLy5wkja6Jfp6hQXlv4QvGMRLRNcLPAogsltpxpywjew',
          Text: 'New York, NY, United States',
        },
      ],
      locationInformation: {},
      idsSelectedDeviceAccess: '',
      utilyEnergySaving: '',
      idsSelectedDevice: '',
      idsSelectedOdu: '',
      idsSelectedDeviceType: '',
      tempLocation: '',
      newDeviceInfo: {},
      previousBcc: '',
      deviceList2: [
        /*{
            id: 2,
            //macId: 'e8fdf8a4773c',
            macId: '34eae7c351e6',
            deviceName: 'Bedroom Thermostat',
            deviceType: 'BCC100',
            isThermostat: true,
            isOn: true,
            mode: 0,
            setPoint: 74,
            heatingSetpoint: 74,
            coolingSetpoint: 79,
            current: 76,
            isAccesories: true,
            accesories: [
              {
                id: 0,
                name: 'Dehumidifier',
                isEnable: 0,
                isOn: false,
              },
              {
                id: 1,
                name: 'Humidifier',
                isEnable: 1,
                isOn: false,
              },
            ],
            isMonitoring: null,
            isOnSchedule: true,
            stage: 3,
            isConnected: true,
            acceleratedHeating: true,
            schedules: [],
          },*/
      ],
    },
  },
  applyMiddleware(thunk),
);

describe('DeadBand screen', () => {
  test('Rendered screen on fahrenheit mode', () => {
    const navState = {params: {isFahrenheit: true}};
    const navigation = {
      getParam: (key, val) => navState?.params[key] ?? val,
    };
    const tree = renderer
      .create(
        <Provider store={realStore}>
          <DeadBandScreen />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Rendered screen on celcius mode', () => {
    const navState = {params: {isFahrenheit: true}};
    const navigation = {
      getParam: (key, val) => navState?.params[key] ?? val,
    };
    const tree = renderer
      .create(
        <Provider store={realStoreCelcius}>
          <DeadBandScreen />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Change deadband', () => {
    const navState = {params: {isFahrenheit: true}};
    const navigation = {
      getParam: (key, val) => navState?.params[key] ?? val,
    };
    const calledFunction = jest.fn();

    const rendered = render(
      <Provider store={realStore}>
        <DeadBandScreen />
      </Provider>,
    );

    const setAdccessSettings = jest.spyOn(
      homeOwnerActions,
      'setAdccessSettings',
    );

    const setDeadBand = rendered.getByTestId('setDeadBand');
    fireEvent(setDeadBand, 'confirm');

    setAdccessSettings();
  });

  test('Render component', () => {
    const navState = {params: {isFahrenheit: true}};
    const navigation = {
      getParam: (key, val) => navState?.params[key] ?? val,
    };
    const calledFunction = jest.fn();

    const rendered = render(
      <Provider store={realStore}>
        <DeadBandScreen />
      </Provider>,
    );
  });
});
