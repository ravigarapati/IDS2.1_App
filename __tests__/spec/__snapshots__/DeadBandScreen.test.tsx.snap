// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeadBand screen Rendered screen on celcius mode 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "space-between",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/header_ribbon.png",
      }
    }
    style={
      Object {
        "height": 8,
        "width": "100%",
      }
    }
  />
  <RCTScrollView>
    <View>
      <View>
        <View
          style={
            Object {
              "padding": 15,
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "width": "100%",
                }
              }
            >
              <View
                accessibilityHint="Activate to open the modal to select the deadband."
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderBottomWidth": 1,
                      "height": 56,
                      "paddingLeft": 10,
                      "paddingRight": 16,
                      "width": "100%",
                    },
                    Object {
                      "backgroundColor": "#EFF1F2",
                    },
                    Object {
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                    },
                  ]
                }
                testID="setDeadBand"
              >
                <View>
                  <Text
                    accessible={true}
                    numberOfLines={0}
                    style={
                      Array [
                        Object {
                          "textAlign": "left",
                        },
                        Object {
                          "color": "#000000",
                        },
                        Object {
                          "fontSize": 12,
                          "lineHeight": 18,
                        },
                        Object {
                          "fontFamily": "BoschSans-Regular",
                        },
                        Object {},
                      ]
                    }
                  >
                    Set to
                  </Text>
                  <Text
                    accessible={true}
                    numberOfLines={0}
                    style={
                      Array [
                        Object {
                          "textAlign": "left",
                        },
                        Object {
                          "color": "#000000",
                        },
                        Object {
                          "fontSize": 16,
                          "lineHeight": 24,
                        },
                        Object {
                          "fontFamily": "BoschSans-Regular",
                        },
                        Object {},
                      ]
                    }
                  >
                    0 °C
                  </Text>
                </View>
                <svg
                  data-file-name="SvgTemperature1"
                  fill="#000"
                />
              </View>
              <Modal
                animationType="slide"
                hardwareAccelerated={false}
                onRequestClose={[Function]}
                transparent={true}
                visible={false}
              />
            </View>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginTop": 33,
              }
            }
          >
            <Text
              accessibilityLabel="Info"
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#00629A",
                    "fontSize": 20,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Bosch-Icon",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
            <Text
              accessibilityLabel="Use the deadband to adjust the minimum value allowed between the heating & cooling set points in Auto mode."
              accessible={true}
              numberOfLines={0}
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "color": "#000000",
                  },
                  Object {
                    "fontSize": 12,
                    "lineHeight": 18,
                  },
                  Object {
                    "fontFamily": "BoschSans-Regular",
                  },
                  Object {
                    "flexShrink": 1,
                    "paddingLeft": 12,
                  },
                ]
              }
            >
              Use the deadband to adjust the minimum value allowed between the heating & cooling set points in Auto mode.

            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`DeadBand screen Rendered screen on fahrenheit mode 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "space-between",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/header_ribbon.png",
      }
    }
    style={
      Object {
        "height": 8,
        "width": "100%",
      }
    }
  />
  <RCTScrollView>
    <View>
      <View>
        <View
          style={
            Object {
              "padding": 15,
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "width": "100%",
                }
              }
            >
              <View
                accessibilityHint="Activate to open the modal to select the deadband."
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderBottomWidth": 1,
                      "height": 56,
                      "paddingLeft": 10,
                      "paddingRight": 16,
                      "width": "100%",
                    },
                    Object {
                      "backgroundColor": "#EFF1F2",
                    },
                    Object {
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                    },
                  ]
                }
                testID="setDeadBand"
              >
                <View>
                  <Text
                    accessible={true}
                    numberOfLines={0}
                    style={
                      Array [
                        Object {
                          "textAlign": "left",
                        },
                        Object {
                          "color": "#000000",
                        },
                        Object {
                          "fontSize": 12,
                          "lineHeight": 18,
                        },
                        Object {
                          "fontFamily": "BoschSans-Regular",
                        },
                        Object {},
                      ]
                    }
                  >
                    Set to
                  </Text>
                  <Text
                    accessible={true}
                    numberOfLines={0}
                    style={
                      Array [
                        Object {
                          "textAlign": "left",
                        },
                        Object {
                          "color": "#000000",
                        },
                        Object {
                          "fontSize": 16,
                          "lineHeight": 24,
                        },
                        Object {
                          "fontFamily": "BoschSans-Regular",
                        },
                        Object {},
                      ]
                    }
                  >
                    0 °F
                  </Text>
                </View>
                <svg
                  data-file-name="SvgTemperature1"
                  fill="#000"
                />
              </View>
              <Modal
                animationType="slide"
                hardwareAccelerated={false}
                onRequestClose={[Function]}
                transparent={true}
                visible={false}
              />
            </View>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginTop": 33,
              }
            }
          >
            <Text
              accessibilityLabel="Info"
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#00629A",
                    "fontSize": 20,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Bosch-Icon",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
            <Text
              accessibilityLabel="Use the deadband to adjust the minimum value allowed between the heating & cooling set points in Auto mode."
              accessible={true}
              numberOfLines={0}
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "color": "#000000",
                  },
                  Object {
                    "fontSize": 12,
                    "lineHeight": 18,
                  },
                  Object {
                    "fontFamily": "BoschSans-Regular",
                  },
                  Object {
                    "flexShrink": 1,
                    "paddingLeft": 12,
                  },
                ]
              }
            >
              Use the deadband to adjust the minimum value allowed between the heating & cooling set points in Auto mode.

            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
